examples/larger-examples/datalog/single-source-shortest-distance.flix

```diff
@@ -0,0 +1,181 @@
+mod ShortestDistance {
+
+    use Down.Down;
+
+    ///
+    /// Returns the shortest distance from `src` to every other node in the
+    /// weighted directed graph `g`.
+    ///
+    pub def distancesFrom(src: t, g: m[(t, t, Int32)]): Map[t, Int32] with Foldable[m], Order[t] = {
```

Comment:
-> sssd

Reply:
In the paper, unfortunately, I always use edges `t, Int32, t` because conceptually the distance is between the two endpoints. Could you refactor? I know its tedious.